### ISSUE TITLE
when DNS fails the component fails to load.

### ISSUE THREAD

**Issue by ausil** [firstof9/openevse#135]
I had an issue with DNS. The visible sign was that the component failed to load in the logs I got 
```
2023-02-07 11:41:40.688 ERROR (MainThread) [custom_components.openevse] Error fetching OpenEVSE (openevse_2) data: Cannot connect to host openevse2.xxxx.xx:80 ssl:default [Name does not resolve]
```
The main visible sign is the following log snippet 
```
Feb 07 15:14:07  homeassistant[438]: 2023-02-07 09:14:07.961 ERROR (MainThread) [custom_components.openevse] Unexpected error fetching OpenEVSE (openevse1).firmware data: 'version'
Feb 07 15:14:07  homeassistant[438]: Traceback (most recent call last):
Feb 07 15:14:07  homeassistant[438]:   File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 239, in _async_refresh
Feb 07 15:14:07  homeassistant[438]:     self.data = await self._async_update_data()
Feb 07 15:14:07  homeassistant[438]:   File "/config/custom_components/openevse/__init__.py", line 242, in _async_update_data
Feb 07 15:14:07  homeassistant[438]:     self._data = await self._manager.firmware_check()
Feb 07 15:14:07  homeassistant[438]:   File "/usr/local/lib/python3.10/site-packages/openevsehttp/__main__.py", line 440, in firmware_check
Feb 07 15:14:07  homeassistant[438]:     _LOGGER.debug("Detected firmware: %s", self._config["version"])
Feb 07 15:14:07  homeassistant[438]: KeyError: 'version'
```

if the name does not resolve or connecting to the OpenEVSE we should return a message up so there is a better idea what went wrong

**Comment by firstof9** [comment]
Restart HA, looks like the library didn't download.

**Comment by ausil** [comment]
Turns out the issue was that one of my DNS servers was playing up and not resolving any internal domains. The error should be handled a bit better

**Comment by ausil** [comment]
I have updated the initial comment with some more details. When I get time I will see if I can fix it

**Comment by firstof9** [comment]
The key error comes from the library, I've since updated it to handle the missing key better.

**Comment by firstof9** [comment]
Should be resolved in `2.1.9`
